corehq/apps/registration/forms.py

```diff
@@ -592,6 +592,10 @@ def clean_email(self):
         if email.lower() in self.excluded_emails:
             raise forms.ValidationError(_("A user with this email address is already in "
                                           "this project or has a pending invitation."))
+        web_user = WebUser.get_by_username(email)
+        if web_user and not web_user.is_active:
+            raise forms.ValidationError(_("A user with this email address is deactivated. "
+                                          "Please reactivate this user first."))
```

Comment:
Not blocking: it'd be a nice UX improvement to turn "reactivate this user" into a link to the edit page for the relevant web user. (Granted, then you have to make sure whatever code catches this error ultimately displays it as HTML.)

Reply:
Thanks Jenny. Usually we don't deactivate user, deactivation and reactivation can be operated by us in Admin>Look up user by email. So having that link might not be very helpful for most customers. For SSO user who get auto-deactivated, they need first add the user back to their identity provider, then ask the user to relogin, I feel this is too long in a error message?

Reply:
I see. If most users won't have access, perhaps we shouldn't tell them to reactivate the user? We have other messages that say to contact support (if that's the appropriate action), or maybe just removing the second sentence would be fine.

Reply:
Yes, removed in 409c134ebfb67cf6f3fdd837c06f313146e526ee ...